salt/minion.py

```diff
@@ -1106,6 +1107,7 @@ def __init__(self, opts, timeout=60, safe=True, loaded_base_name=None, io_loop=N
         self.ready = False
         self.jid_queue = [] if jid_queue is None else jid_queue
         self.periodic_callbacks = {}
+        self.thread_pool = ThreadPoolExecutor(max_workers=4)
```

Comment:
In the spirit of Salt, this should be configurable (not hard-coded to 4).

Reply:
Yeah, I kind of agree... I was thinking of reusing the `process_count_max` option, but that would probably be confusing since users would expect that one to be taken in consideration only when multiprocessing is enabled. On the other side, this doesn't really bring any real parallelism. It's just letting you run long and short jobs in pseudo-parallel so that you don't have to wait for the long job to finish. So I'd expect that this setting should not be touched very often.

Reply:
Since the documentation for process_count_max says: "Limit the maximum amount of processes or **threads**", I went ahead and reused that config option.